### PR TITLE
ls: show/hide control chars

### PR DESCRIPTION
**Pull request by tertsdiepraam**
Implements the flags:
- `-q` & `--hide-control-chars`
- `--show-control-chars`

The former is now the default and replaces any control character with a `?`, the second just shows them as is.

One thing I couldn't figure out was this line from the GNU docs for the `-q` flag:
> This is the default if the output is a terminal and the program is `ls`.

In what circumstances would the program not be `ls` and how could we perform the same check? If anyone knows how to do this, please let me know and I'll update the PR, otherwise I'll make an issue about this.

### CODE REVIEW

File: src/uu/ls/src/ls.rs [hunk 1]
```diff
@@ -366,24 +367,36 @@ impl Config {
             })
             .or_else(|| termsize::get().map(|s| s.cols));
 
+        let show_control = if options.is_present(options::HIDE_CONTROL_CHARS) {
+            false
```
Comment (sylvestre) [inline src/uu/ls/src/ls.rs:371]: it is possible to add a test for this case too?

**Review by sylvestre — COMMENTED** [review]
well done for the tests :)

File: src/uu/ls/src/ls.rs (hunk 1 above)
Comment (tertsdiepraam) [reply src/uu/ls/src/ls.rs:371]: Ah of course, forgot about that because it is the default, but I added some more tests now

Comment (sylvestre) [reply src/uu/ls/src/ls.rs:371]: as always, we forgot to test the most obvious stuff :)

**Review by sylvestre — APPROVED** [review]

**Pull request merged**